File: JavaScript/commonFactors.js
const commonFactors = (n1, n2) => {
  let greater;
  let res = [];
  n1 > n2 ? (greater = n1) : (greater = n2);
  for (let i = 0; i < greater; i++) {
    n1 % i == 0 && n2 % i == 0 ? res.push(i) : {};
  }
  return res;
};

console.log(commonFactors(50, 30));
console.log(commonFactors(8, 4));
console.log(commonFactors(4, 8));

File: Python/common_factors.py
def common_factors(n1, n2):
    lst = []
    if n1 > n2:
        greater = n1
    else:
        greater = n2
    for i in range(1,greater):
        if n1 % i == 0 and n2 % i == 0:
            lst.append(i)
    return lst

print(common_factors(50,30))
print(common_factors(8,4))
print(common_factors(4,8))
